lib/matrix_sdk/request.ex

```diff
@@ -375,6 +485,93 @@ defmodule MatrixSDK.Request do
     }
   end
 
+  @doc """
+  Returns a `%Request{}` struct used to request validation tokens when authenticating for `change_password`. 
+
+  ## Examples
+
+      iex> MatrixSDK.Request.password_email_token("https://matrix.org", "secret", "maurice@moss.yay", 1)
+      %MatrixSDK.Request{
+        base_url: "https://matrix.org",
+        body: %{client_secret: "secret", email: "maurice@moss.yay", send_attempt: 1},
+        headers: [],
+        method: :post,
+        path: "/_matrix/client/r0/account/password/email/requestToken"
+      }
+  """
+  @spec password_email_token(base_url, binary, binary, pos_integer, map) :: t
+  def password_email_token(base_url, client_secret, email, send_attempt, opts \\ %{}) do
+    body =
+      %{}
+      |> Map.put(:client_secret, client_secret)
+      |> Map.put(:email, email)
+      |> Map.put(:send_attempt, send_attempt)
+      |> Map.merge(opts)
+
+    %__MODULE__{
+      method: :post,
+      base_url: base_url,
+      path: "/_matrix/client/r0/account/password/email/requestToken",
+      body: body
+    }
+  end
+
+  @doc """
+  Returns a `%Request{}` struct used to request validation tokens when authenticating for `change_password`. 
+
+  ## Examples
+
+      iex> MatrixSDK.Request.password_msisdn_token("https://matrix.org", "secret", "GB", "07700900001", 1)
+      %MatrixSDK.Request{
+        base_url: "https://matrix.org",
+        body: %{client_secret: "secret", country: "GB", phone_number: "07700900001", send_attempt: 1},
+        headers: [],
+        method: :post,
+        path: "/_matrix/client/r0/account/password/msisdn/requestToken"
+      }
+  """
+  @spec password_msisdn_token(base_url, binary, binary, binary, pos_integer, map) :: t
+  def password_msisdn_token(base_url, client_secret, country, phone, send_attempt, opts \\ %{}) do
+    body =
+      %{}
+      |> Map.put(:client_secret, client_secret)
+      |> Map.put(:country, country)
+      |> Map.put(:phone_number, phone)
+      |> Map.put(:send_attempt, send_attempt)
+      |> Map.merge(opts)
+
+    %__MODULE__{
+      method: :post,
+      base_url: base_url,
+      path: "/_matrix/client/r0/account/password/msisdn/requestToken",
+      body: body
+    }
+  end
+
+  @doc """
+  Returns a `%Request{}` struct used to deactivate a user's account. 
+
+  ## Example
+
+      iex> MatrixSDK.Request.deactivate_account("https://matrix.org", "token")
+      %MatrixSDK.Request{
+        base_url: "https://matrix.org",
+        body: %{},
+        headers: [{"Authorization", "Bearer token"}],
+        method: :post,
+        path: "/_matrix/client/r0/account/deactivate"
+      }
+  """
+  @spec deactivate_account(base_url, binary, map) :: t
+  def deactivate_account(base_url, token, opts \\ %{}),
```

Comment:
```suggestion
  def deactivate_account(base_url, token, auth, opts \\ %{}),
```

I think auth is a required parameter here because of the type field.
https://matrix.org/docs/spec/client_server/latest#post-matrix-client-r0-account-deactivate

At least you do that with account_add_3pid. The spec seems to treat the auth param equally in both endpoints.

Granted the spec is a bit ambiguous.

Reply:
Actually I wonder if it's necessary in both endpoints at all—it might be optional, I'll investigate tomorrow.

Reply:
I'm inclined to treat `auth` as optional in both endpoints, given the example for the `account_add_3pid` endpoint. The auth API being interactive, the server response would make that explicit to the user in any case.